solar_system_wave3.rb

```diff
@@ -0,0 +1,109 @@
+class SolarSystem
+
+  def initialize(planets)
+    @planets = planets
+  end
+
+  attr_reader :planets
+
+  def puts_self
+    puts self
+  end
+
+  def self.planets
+    return @planets
+  end
+end
+
+class Planet
+
+  attr_accessor :name, :order, :radius, :moons
+
+  def initialize(name, order, radius, moons)
+    @name = name
+    @order = order
+    @radius = radius
+    @moons = moons
+  end
+
+  def puts_self
+    puts self
+  end
+
+  def self.name
+    return @name
+  end
+
+  # instance method
+  def contents
+    return "  #{order}. #{name}"
+  end
+
+  def self.contents
+    return @contents
+  end
+
+  def summary
+    return "  #{name} - Radius: #{radius}km. Number of Moons: #{moons}."
+  end
+
+  def self.summary
+    return @summary
+  end
+
+end
+
+# input planets
+  planet_arr = [
+    Planet.new("Mercury", 1, 2440, 0),
+    Planet.new("Venus", 2, 6052, 0),
+    Planet.new("Earth", 3, 6371, 1),
+    Planet.new("Mars", 4, 3390, 2),
+    Planet.new("Jupiter", 5, 69911, 2),
+    Planet.new("Saturn", 6, 58232, 2),
+    Planet.new("Uranus", 7, 25362, 2),
+    Planet.new("Neptune", 8, 24622, 2),
+  ]
+
+#outputs planets
+puts
+print "Welcome to this Solar System directory! Here is a list of planets: "
+puts
+planet_arr.each_with_index do |planet, i|
+  puts "#{i+1}. #{planet.name}"
+end
+
+puts
+print "Type LEARN to learn about a planet. Type NEW to add a new planet: "
+selection = gets.chomp.upcase!
+if selection == "LEARN"
+  print "Please type the number of the planet you'd like to learn about: "
+  learn_option = gets.chomp.to_i
+  counter = 0
+  until learn_option >= 0 &&  learn_option <= planet_arr.length
+    counter += 1
+    if counter == 3
+      puts "Too many incorrect attempts. Goodbye"
+      exit
+    else
+      print "You did not type a valid number. Please try again: "
+      learn_option = gets.chomp.to_i
+    end
+  end
+  puts planet_arr[learn_option-1].summary
```

Comment:
starting at line 81, you have a small bug!

when `gets.chomp.to_i` executes on line 81, if I type in a non-integer (like "asdlfkjasdf"), then the value of `learn_option` will default to `0`.

Because `learn_option` is 0, it will skip over the `until` block on line 83, and then jump straight to line 93.  if `learn_option` is 0, then `planet_arr[0-1]` will mean `planet_arr[-1]`, which will always choose the last element in the array.